modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java

```diff
@@ -1306,6 +1347,147 @@ public void testCorrectPartitionRecoveryOnSnapshot() throws InterruptedException
         }
     }
 
+    @Test
+    public void tableRecoveryOnMultipleRestartingNodesTest() throws InterruptedException {
+        int euIdx = 0;
+        int usIdx = 1;
+        int ruIdx = 2;
+        int jpIdx = 3;
+
+        IgniteImpl nodeEu = startNode(euIdx, configurationString(euIdx, "{ region.attribute = \"EU\" }"));
+        IgniteImpl nodeUs = startNode(usIdx, configurationString(usIdx, "{ region.attribute = \"US\" }"));
+
+        startNode(ruIdx, configurationString(ruIdx, "{ region.attribute = \"RU\" }"));
+        startNode(jpIdx, configurationString(jpIdx, "{ region.attribute = \"JP\" }"));
+
+        WatchListenerInhibitor nodeEuInhibitor = WatchListenerInhibitor.metastorageEventsInhibitor(nodeEu);
+
+        String tableName = "TEST";
+        String startFilter = "$[?(@.region == \"EU\" || @.region == \"US\")]";
+        String alteredFilter = "$[?(@.region == \"RU\" || @.region == \"JP\")]";
+
+        try (Session session = nodeUs.sql().createSession()) {
+            session.execute(null, String.format("CREATE ZONE IF NOT EXISTS ZONE_%s WITH REPLICAS=%d, PARTITIONS=%d, "
+                            + "DATA_NODES_FILTER='%s'", tableName, 3, 1, startFilter));
+
+            nodeEuInhibitor.startInhibit();
+
+            session.execute(null, "CREATE TABLE " + tableName
+                    + "(id INT PRIMARY KEY, name VARCHAR) WITH PRIMARY_ZONE='ZONE_" + tableName + "';");
+        }
+
+        int tableId = tableId(nodeUs, tableName);
+
+        removeStableAssignments(usIdx, tableId, 0);
```

Comment:
It'll be tricky to support the invariant that to this point stable assignments were written to meta storage. I'd rather add waitForCondidition that will ensure that assignments were written before removing them.

Reply:
It creates a fake data for partial nodes. I have described this in javadoc.

Reply:
1. The provided javadoc still seems unclear to me. Could you please rephrase it? Maybe explicit test steps with <li> will help.
2. Despite the fact that currently create table includes synchronous assignments update it's not the part of the contract. And thus it'll be extremely tricky to analyse test failures caused by the race between assignments update and assignments remove. So that, prior to removeStableAssignments I'd rather add waitForCondidition that will ensure that assignments were written.